include/network/networkinterfacemanager.hpp

```diff
@@ -7,10 +7,106 @@
 #ifndef NETWORKINTERFACEMANAGER_HPP_
 #define NETWORKINTERFACEMANAGER_HPP_
 
+#include <optional>
+#include <string>
+
+#include <sys/socket.h>
+
 #include <aos/sm/networkmanager.hpp>
+namespace aos::sm::networkmanager {
+
+/**
+ * Link attributes.
+ */
+struct LinkAttrs {
```

Comment:
Probably should be in separate file as it is in `aos::sm::networkManager` namespace. Shouldn't it be part of SM instead? Looks like some design issue need to discuss.